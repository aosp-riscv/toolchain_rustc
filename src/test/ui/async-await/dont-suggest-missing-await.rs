// edition:2018

// This test ensures we don't make the suggestion in bodies that aren't `async`.

<<<<<<< HEAD   (086005 Importing rustc-1.38.0)
#![feature(async_await)]

=======
>>>>>>> BRANCH (8cd2c9 Importing rustc-1.39.0)
fn take_u32(x: u32) {}

async fn make_u32() -> u32 {
    22
}

async fn dont_suggest_await_in_closure() {
    || {
        let x = make_u32();
        take_u32(x)
        //~^ ERROR mismatched types [E0308]
    };
}

fn main() {}
