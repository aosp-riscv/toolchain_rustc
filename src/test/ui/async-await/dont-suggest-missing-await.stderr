error[E0308]: mismatched types
<<<<<<< HEAD   (086005 Importing rustc-1.38.0)
  --> $DIR/dont-suggest-missing-await.rs:16:18
=======
  --> $DIR/dont-suggest-missing-await.rs:14:18
>>>>>>> BRANCH (8cd2c9 Importing rustc-1.39.0)
   |
LL |         take_u32(x)
   |                  ^ expected u32, found opaque type
   |
   = note: expected type `u32`
              found type `impl std::future::Future`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
